### PR TITLE
Feat/chat

### PR DESCRIPTION
**Pull request by nmorrow11**
This PR adds the chat feature, and includes adding supporting components (`ChatView`, `MessageInput`, and `Messages`). The chat feature uses socket.io to emit messages from a client, receive them on the server, and emit to all the clients. The clients then update their component to reflect the new message.

A message contains `body`, `time`, `username`, and `color` - which is generated from their unique socket ID.

### CODE REVIEW

File: client/src/components/Room/ChatView.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+class ChatView extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      messages: [],
+    };
+    this.sendMessage = this.sendMessage.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
```
Comment (aricalves) [inline client/src/components/Room/ChatView.jsx]: GG Brooke

File: server/index.js [hunk 2]
```diff
@@ -114,4 +113,15 @@ roomSpace.on('connection', (socket) => {
...
     }
   });
+
+  socket.on('emitMessage', (message) => {
+    let sum = 0;
+    for (let i = 0; i < 3; i ++) {
+      sum += message.userName.charCodeAt(i);
+    }
+    const colors = ['#ffb3ba', '#ffd2b3', '#fff8b3', '#baffb3', '#bae1ff', '#e8baff'];
+    const userColor = colors[(sum % colors.length)];
```
Comment (aricalves) [inline server/index.js]: NICE

File: client/src/components/Room/RoomView.jsx [hunk 3]
```diff
@@ -6,8 +6,9 @@ import axios from 'axios';
 import VideoPlayer from './VideoPlayer';
 import Playlist from './Playlist';
 import Search from './Search';
+import ChatView from './ChatView';
 
-// const socket = io.connect(window.location.hostname);
+const socket = io.connect('localhost:5000');
```
Comment (aricalves) [inline client/src/components/Room/RoomView.jsx]: Why change from `location.hostname`?

File: client/src/components/Room/MessageInput.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    };
+    this.setMessage = this.setMessage.bind(this);
+    this.enterMessage = this.enterMessage.bind(this);
+  }
+
+  setMessage(event) {
+    const pressedEnter = event.key === 'Enter';
+    if (pressedEnter) {
+      let time = new Date();
+      time = time.toLocaleTimeString();
```
Comment (kk1024) [inline client/src/components/Room/MessageInput.jsx]: make line 17 and 18 one line?

**Pull request closed without merging**